src/EditorFeatures/CSharpTest/GenerateComparisonOperators/GenerateComparisonOperatorsTests.cs

```diff
@@ -0,0 +1,448 @@
+        return left.CompareTo(right) > 0;
+    }
+
+    public static bool operator <=(C left, C right)
+    {
+        return left.CompareTo(right) <= 0;
+    }
+
+    public static bool operator >=(C left, C right)
+    {
+        return left.CompareTo(right) >= 0;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateComparisonOperators)]
+        public async Task TestMissingWithoutCompareMethod()
+        {
+            await TestMissingAsync(
+@"
+using System;
+
+class C : IComparable<C>
+{
+[||]
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateComparisonOperators)]
+        public async Task TestMissingWithUnknownType()
+        {
+            await TestMissingAsync(
+@"
+using System;
+
+class C : IComparable<Goo>
+{
+    public int CompareTo(Goo g) => 0;
+
+[||]
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateComparisonOperators)]
+        public async Task TestMissingWithAllExistingOperators()
+        {
+            await TestMissingAsync(
+@"
+using System;
+
+class C : IComparable<C>
+{
+    public int CompareTo(C c) => 0;
+
+    public static bool operator <(C left, C right)
+    {
+        return left.CompareTo(right) < 0;
+    }
+
+    public static bool operator >(C left, C right)
+    {
+        return left.CompareTo(right) > 0;
+    }
+
+    public static bool operator <=(C left, C right)
+    {
+        return left.CompareTo(right) <= 0;
+    }
+
+    public static bool operator >=(C left, C right)
+    {
+        return left.CompareTo(right) >= 0;
+    }
+
+[||]
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateComparisonOperators)]
+        public async Task TestWithExistingOperator()
+        {
+            await TestInRegularAndScript1Async(
+@"
+using System;
+
+class C : IComparable<C>
+{
+    public int CompareTo(C c) => 0;
+
+    public static bool operator <(C left, C right)
+    {
+        return left.CompareTo(right) < 0;
+    }
+
+[||]
+}",
+@"
+using System;
+
+class C : IComparable<C>
+{
+    public int CompareTo(C c) => 0;
+
+    public static bool operator <(C left, C right)
+    {
+        return left.CompareTo(right) < 0;
+    }
+
+    public static bool operator >(C left, C right)
+    {
+        return left.CompareTo(right) > 0;
+    }
+
+    public static bool operator <=(C left, C right)
+    {
+        return left.CompareTo(right) <= 0;
+    }
+
+    public static bool operator >=(C left, C right)
+    {
+        return left.CompareTo(right) >= 0;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateComparisonOperators)]
+        public async Task TestMultipleInterfaces()
+        {
+            await TestInRegularAndScript1Async(
+@"
+using System;
+
+class C : IComparable<C>, IComparable<int>
+{
+    public int CompareTo(C c) => 0;
+    public int CompareTo(int c) => 0;
+
+[||]
+}",
```

Comment:
Can we add a test for also adding operators for `IComparable<C>`? Do we support adding for both?